packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs

```diff
@@ -175,5 +187,30 @@ private ParameterValidationType GetParameterValidation()
 
             return ParameterValidationType.AssertNotNull;
         }
+
+        private static ValueExpression? GetParameterInitializationValue(InputParameter inputParameter)
```

Comment:
I don't think this is something that the Parameter can know since it depends on the context of the method or ctor its a parameter for.

Feels like InitializationValue is something that should be passed in and defaulted to null

Reply:
I've defaulted InitializationValue to null and moved this method to the client provider.